mm2src/mm2_main/src/lp_stats.rs

```diff
@@ -170,16 +170,20 @@ struct Mm2VersionRes {
     nodes: HashMap<String, String>,
 }
 
-fn process_get_version_request(ctx: MmArc) -> Result<Option<Vec<u8>>, String> {
+fn process_get_version_request(ctx: MmArc) -> Result<Vec<u8>, String> {
     let response = ctx.mm_version().to_string();
-    let encoded = try_s!(encode_message(&response));
-    Ok(Some(encoded))
+    encode_message(&response).map_err(|e| e.to_string())
 }
 
-pub fn process_info_request(ctx: MmArc, request: NetworkInfoRequest) -> Result<Option<Vec<u8>>, String> {
-    log::debug!("Got stats request {:?}", request);
+fn process_get_peer_utc_timestamp_request() -> Result<Vec<u8>, String> {
+    let timestamp = common::get_utc_timestamp();
+    encode_message(&timestamp).map_err(|e| e.to_string())
```

Comment:
I think that explanation alone makes it more complicated than it should be. I didn't think it would be this confusing :thinking:. I would rather cast it to `u64` before sending it instead of explaining it like "we do this because of encoding numbers..."

Reply:
me too tbh, i favor explicit casting all the way :/